apps-devstg/us-east-1/k8s-eks/identities/ids_external_secrets.tf

```diff
@@ -0,0 +1,87 @@
+#
+# External-Secrets Roles & Policies
+#
+module "role_external_secrets" {
+  source = "github.com/binbashar/terraform-aws-iam.git//modules/iam-assumable-role-with-oidc?ref=v5.2.0"
+
+  create_role  = true
+  role_name    = "${local.environment}-external-secrets"
+  provider_url = replace(data.terraform_remote_state.eks-cluster.outputs.cluster_oidc_issuer_url, "https://", "")
+
+  role_policy_arns = [
+    aws_iam_policy.external_secrets_secrets_manager.arn,
+    aws_iam_policy.external_secrets_parameter_store.arn
+  ]
+  oidc_fully_qualified_subjects = [
+    "system:serviceaccount:external-secrets:external-secrets"
+  ]
+
+  tags = local.tags_external_secrets
+}
+
+resource "aws_iam_policy" "external_secrets_secrets_manager" {
+  name        = "${local.environment}-external-secrets-secrets-manager"
+  description = "External-secrets permissions on Secrets Manager"
+  tags        = local.tags_external_secrets
+  policy      = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "secretsmanager:GetResourcePolicy",
+        "secretsmanager:GetSecretValue",
+        "secretsmanager:DescribeSecret",
+        "secretsmanager:ListSecretVersionIds"
+      ],
+      "Resource": [
+        "arn:aws:secretsmanager:${var.region}:${var.accounts.apps-devstg.id}:secret:/k8s-eks/*"
+      ]
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "kms:Decrypt",
+        "kms:DescribeKey"
+      ],
+      "Resource": [
+        "arn:aws:kms:${var.region}:${var.accounts.apps-devstg.id}:key/*"
```

Comment:
I think this way is too permissive. We should only grant this permission on the specific key that is used to encrypt the secrets.  In our case I believe we use the default key of the account, however, on Client implementations we may want to advise them to use a different key than the one they use to protect other data (e.g. objects in buckets, CloudTrail logs, VPC Flow logs, and so on).

Reply:
Added a datasource and addressed this issue!